de.fu_berlin.inf.dpp.intellij/src/de/fu_berlin/inf/dpp/intellij/project/SharedResourcesManager.java

```diff
@@ -146,21 +138,22 @@ public void receive(FolderCreatedActivity activity) {
         @Override
         public void receive(FolderDeletedActivity activity) {
             try {
-                handleFolderActivity(activity);
+                handleFolderDeletion(activity);
             } catch (IOException e) {
                 LOG.error("Failed to execute activity: " + activity, e);
             }
         }
     };
 
-    private void handleFileActivity(FileActivity activity) throws IOException {
+    private void handleFileActivity(
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Move this method into the anonymous class declared at line 105. ([squid:S3398](https://sonarcloud.io/organizations/saros/rules#rule_key=squid%3AS3398))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=saros_complete&pullRequest=277&issues=AWagwjc6lHK86HcoyTJF&open=AWagwjc6lHK86HcoyTJF)

[ISSUE_KEY]: AWagwjc6lHK86HcoyTJF